src/platform_impl/linux/wayland/seat/pointer/mod.rs

```diff
@@ -0,0 +1,321 @@
+//! All pointer related handling.
+
+use std::cell::{Cell, RefCell};
+use std::rc::{Rc, Weak};
+
+use sctk::reexports::client::protocol::wl_pointer::WlPointer;
+use sctk::reexports::client::protocol::wl_seat::WlSeat;
+use sctk::reexports::client::protocol::wl_surface::WlSurface;
+use sctk::reexports::client::Attached;
+use sctk::reexports::protocols::unstable::relative_pointer::v1::client::zwp_relative_pointer_manager_v1::ZwpRelativePointerManagerV1;
+use sctk::reexports::protocols::unstable::relative_pointer::v1::client::zwp_relative_pointer_v1::ZwpRelativePointerV1;
+use sctk::reexports::protocols::unstable::pointer_constraints::v1::client::zwp_pointer_constraints_v1::{ZwpPointerConstraintsV1, Lifetime};
+use sctk::reexports::protocols::unstable::pointer_constraints::v1::client::zwp_confined_pointer_v1::ZwpConfinedPointerV1;
+
+use sctk::seat::pointer::{ThemeManager, ThemedPointer};
+
+use crate::event::ModifiersState;
+use crate::platform_impl::wayland::event_loop::WinitState;
+use crate::window::CursorIcon;
+
+mod data;
+mod handlers;
+
+use data::PointerData;
+
+/// A proxy to Wayland pointer, which serves requests from a `WindowHandle`.
+pub struct WinitPointer {
+    pointer: ThemedPointer,
```

Comment:
Yeah, called WlPointer and it lives in wayland_client, Themed pointer is sctk thing that allows you to change cursor to some common cursors, since cursor drawing is done in clients on Wayland. So, Theme pointer uses cursors from ThemeManager that loads your system XCursor theme.

Reply:
That's about what I expected, yeah. It does theme a little strange, but I guess it makes sense.